### PR TITLE
backend: User auth fixups

### PR DESCRIPTION
**Pull request by wgreenberg**
Some changes and fixups to user auth, as well as improvements to server-wide config and static file serving

### CODE REVIEW

File: backend/src/extractors/auth.rs [hunk 1]
```diff
@@ -50,7 +52,12 @@ async fn validate_and_fetch_uid(token: &str) -> Result<String, Error> {
     let jwk = jwks.find(&kid).expect("Specified key not found in set");
     match validate(token, jwk, validations) {
         Ok(jwt) => match jwt.claims.get("sub") {
-            Some(value) => Ok(value.to_string()),
+            Some(value) => {
+                let token =  value.as_str()
```
Comment (klautcomputing) [inline backend/src/extractors/auth.rs]: You either didn't `cargo fmt` this, or we have a different config :slightly_smiling_face:

File: backend/src/model/user.rs [hunk 2]
```diff
@@ -5,18 +5,53 @@ use diesel::{result::Error, Identifiable, Insertable, QueryDsl, Queryable};
 use diesel_async::RunQueryDsl;
 use serde::{Deserialize, Serialize};
 
-#[derive(Queryable, Identifiable, Insertable, Serialize, Deserialize)]
-#[primary_key(uid)]
+const MAX_USERNAME_LENGTH: usize = 20;
```
Comment (klautcomputing) [inline backend/src/model/user.rs:8]: Might be a bit short, but it's also something we can easily change later.

File: backend/migrations/2023-02-24-030452_create_users/up.sql [hunk 3]
```diff
@@ -1,6 +1,6 @@
 -- Your SQL goes here
 CREATE TABLE users (
     uid text primary key not null,
-    username text not null,
+    username text not null unique,
```
Comment (klautcomputing) [inline backend/migrations/2023-02-24-030452_create_users/up.sql:4]: why are you setting a max length in the model  but not for the actual database field?

**Review by klautcomputing — CHANGES_REQUESTED** [review]
Overall this looks great just some small things!

File: backend/src/config.rs [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+impl ServerConfig {
+    pub fn from_env() -> Result<ServerConfig, VarError> {
+        Ok(ServerConfig {
+            database_uri: env::var("DATABASE_URI")?,
+            firebase_jwt_issuer: env::var("FIREBASE_JWT_ISSUER")?,
+            firebase_jwt_authority: env::var("FIREBASE_JWT_AUTHORITY")?,
+            static_files_path: env::var("STATIC_FILES_PATH")?.into(),
+        })
+    }
+}
```
[the remark this answers is not in the document]
Comment (wgreenberg) [reply backend/src/config.rs]: ah yeah, i recently switched to vscode (on linux ironically) which doesn't do this by default. fixed!

File: backend/migrations/2023-02-24-030452_create_users/up.sql (hunk 3 above)
Comment (wgreenberg) [reply backend/migrations/2023-02-24-030452_create_users/up.sql:4]: ah, good idea!

**Review by klautcomputing — APPROVED** [review]

**Pull request merged**